### PR TITLE
Add env vars

### PR DESCRIPTION
**Pull request by bsansouci**
This PR adds a new flag `envvars` which allows you to add environment variables specific to a target.
This is used in [reglexampleproject](https://github.com/bsansouci/reglexampleproject) to pass an env var to [matchenv](https://github.com/bsansouci/matchenv).

### CODE REVIEW

File: src/ocamlDep.re [hunk 1]
```diff
@@ -19,16 +19,19 @@ let ocamlDep source::source target::target => {
...
   let extraFlags = target.flags.dep;
+  let envvars = target.flags.envvars;
   /* betterError doesn't support bucklescript yet */
   let berror = target.engine == "bucklescript" ? "" : "| berror";
 
   /** seems like refmt intelligently detects source code type (re/ml) */
   let getDepAction () =>
     bashf
-      "ocamlfind ocamldep -pp refmt %s -ml-synonym .re -mli-synonym .rei -modules -one-line %s %s %s 2>&1 %s; (exit ${PIPESTATUS[0]})"
+      ignore_stderr::true
```
Comment (vramana) [inline src/ocamlDep.re]: Why do we want to ignore stderr here? Can you leave a comment in the code as well?

Comment (bsansouci) [reply src/ocamlDep.re]: This PR depends on https://github.com/reasonml/rebel/pull/90 that I just updated.

File: src/ocamlDep.re [hunk 2]
```diff
@@ -19,16 +19,18 @@ let ocamlDep source::source target::target => {
   let flag = isInterface source ? "-intf" : "-impl";
   let ppx = target.engine == "bucklescript" ? "-ppx bsppx.exe" : "";
   let extraFlags = target.flags.dep;
+  let envvars = target.flags.envvars;
   /* betterError doesn't support bucklescript yet */
-  let berror = target.engine == "bucklescript" ? "" : "| berror";
+  let berror = target.engine == "bucklescript" ? "" : "2>&1| berror";
```
Comment (jordwalke) [inline src/ocamlDep.re:24]: Oh, another thing I realized is that there was no way to change `berror` to `refmterr`! I think we should add a config field, don't you? (Just to put the name of the binary `berror`/`refmterr`?

(This is totally unrelated to this diff).

File: src/utils.re [hunk 3]
```diff
@@ -259,6 +259,11 @@ let parseTarget t => {
     | Some (`String f) => f
     | _ => ""
     };
+  let envvars =
```
Comment (jordwalke) [inline src/utils.re:262]: Does this print them in the form `X=y` somehow?

**Review by jordwalke — APPROVED** [review]
Thank you!
